src/daemon/daemon_main.cpp

```diff
@@ -47,6 +49,20 @@ sigset_t make_and_block_signals(std::vector<int> sigs)
     return sigset;
 }
 
+void set_server_permissions(const std::string& server_address)
+{
+    auto group = getgrnam("adm");
```

Comment:
I think it should check if the server_address starts with "unix:" before attempting the rest.